cmd/cloud-controller-manager/app/core.go

```diff
@@ -98,12 +101,23 @@ func startRouteController(ctx *cloudcontrollerconfig.CompletedConfig, cloud clou
 		klog.Warning("configure-cloud-routes is set, but cloud provider does not support routes. Will not configure cloud provider routes.")
 		return nil, false, nil
 	}
-	var clusterCIDR *net.IPNet
-	var err error
+	clusterCIDRs := make([]*net.IPNet, 0)
 	if len(strings.TrimSpace(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR)) != 0 {
-		_, clusterCIDR, err = net.ParseCIDR(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR)
-		if err != nil {
-			klog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, err)
+		clusterCIDRList := strings.Split(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, ",")
+		for idx, thisCIDR := range clusterCIDRList {
+			if idx > 0 && !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
+				klog.Warningf("multiple cluster cidrs were supplied but dual stack is not enabled, ignoring all but first")
+				break
+			}
+			if idx > 1 {
+				klog.Warningf("dual stacks enabled but more than 2 cidrs were supplied. ignoring all but first two cidrs")
+				break
+			}
+			_, clusterCIDR, err := net.ParseCIDR(thisCIDR)
+			if err != nil {
+				klog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, err)
+			}
+			clusterCIDRs = append(clusterCIDRs, clusterCIDR)
```

Comment:
seems like you really want `map[family]IPNet` and to check thatthe family was not already defined